app/controllers/provider/admin/cms/sections_controller.rb

```diff
@@ -64,11 +64,7 @@ def destroy
   protected
 
   def section_params
-    params[:cms_section].dup.tap do |params|
-      params[:parent] = if id = params[:parent_id]
-                          current_account.sections.find(id)
-                        end
-    end
+    params.require(:cms_section).permit(:parent_id, :title, :public, :partial_path)
```

Comment:
This change is a bit dangerous, because it allows setting a section that belongs to a different provider.
1. It's only possible when hacking the HTML
2. But it may lead to WTF moments, and in general crossing the tenant borders is not good.

Reply:
You're right, and now I wonder if this could happen also from the API endpoint. I'll take a look. Thanks

Reply:
No, I think in the API the logic is correct.

Reply:
The bug was also present in API endpoints for both Files and Sections. I solved it by adding a validation to the models. I thought it was a better place than controllers because it's in fact a validation problem and I don't think it's good to repeat the same logic (manually filtering between current provider sections) in every controller. This is a summary of what I found:

- Sections:
  - In both API and UI it was accepting any value, even non-existing section IDs
  - It doesn't fail in master, from the API (where the code to manually filter parents by provider is not present), because the `parent_id` parameter is not in the `attr_accessible` list and thus always ignored.
- Files:
  - In both API and UI it was accepting other providers sections, but not non-existing sections. Trying to set an non-existing section returned an error about section being blank.
  - It doesn't fail in master, from the API, due to some Rails black magic that makes every non-existing or existing-but-not-same-provider sections return a "Section can't be blank" error. For some reason, removing `attr_accessible` for the model caused sections belonging to other providers to be accepted, but non-existing sections still return the "Blank section" error. ¯\\_(ツ)_/¯

For templates there's no problem because `TemplateService` contains the logic to filter the given section.